examples/catalog/src/menu/Menu.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import Drawer, {
+  DrawerContent,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerSubtitle,
+} from '@material/react-drawer';
+import List, {
+  ListItem,
+  ListItemGraphic,
+  ListItemText,
+} from '@material/react-list';
+import MaterialIcon from '@material/react-material-icon';
+
+import {MenuContext, UserContext} from '../Context';
+import {Menus} from '../Router';
+
+export const Menu = () => (
+  <MenuContext.Consumer>
+    {({isOpen, setOpen, selectedIndex, setSelectedIndex}) => (
+      <UserContext.Consumer>
+        {({name, account}) => (
+          <Drawer modal open={isOpen} onClose={() => setOpen(false)}>
```

Comment:
please move onClose to its own method outside of the Drawer

Reply:
@TroyTae did you have a reason for not wanting to do this?

Reply:
Oh where is my comment?
I replied like this:
```
Can you tell me what is the benefit when we split this method to outside?
I written like this because I want to leave it as expression.
To split method, we need to make it a statement.
```

Reply:
Oh weird we lost that comment.

So if you keep it as an inline function like that, it will rerender every time. If you move it to a named function outside of the render method, it will not rerender because of the method.

Reply:
> If you move it to a named function outside of the render method, it will not rerender because of the method.

Can you explain to me about this mean?
I knew that React's Virtual Dom will detect what has changed and reflect only changed part.

Reply:
It detects an anonymous function as a new function each render. So it should instead be like this:

```
const myFunc = () => {
  // do something
}

...


render() {
  return (
    <MyComp onSomething={myFunc} />
  );
}
```

Reply:
Ah I see, I'm working on it!

Reply:
ah gotcha...hmm I guess we can ignore this one for now